migrations/back_populate_uuids_plate_barcodes.py

```diff
@@ -0,0 +1,202 @@
+                        counter_mongo_update_failures += 1
+
+                except Exception as e:
+                    counter_mongo_update_failures += 1
+                    LOGGER.critical("Failed to update sample in Mongo for mongo id " f"{sample_doc[FIELD_MONGODB_ID]}")
+                    LOGGER.exception(e)
+
+    LOGGER.info(f"Count of successful Mongo updates = {counter_mongo_update_successes}")
+    LOGGER.info(f"Count of failed Mongo updates = {counter_mongo_update_failures}")
+
+
+def update_mongo_sample(samples_collection: Collection, sample_doc: SampleDoc) -> bool:
+    """Updates a sample in the Mongo samples collection
+
+    Arguments:
+        config {Config} -- application config specifying database details
+        sample_doc {SampleDoc} -- the sample document whose fields should be updated
+
+    Returns:
+        bool -- whether the updates completed successfully
+    """
+    try:
+        mongo_sample = samples_collection.find_one_and_update(
+            filter={
+                FIELD_MONGODB_ID: sample_doc[FIELD_MONGODB_ID],
+            },
+            update={
+                "$set": {
+                    FIELD_LH_SAMPLE_UUID: sample_doc[FIELD_LH_SAMPLE_UUID],
+                    FIELD_UUID_UPDATED: True,
+                    FIELD_UPDATED_AT: sample_doc[FIELD_UPDATED_AT],
+                }
+            },
+        )
+
+        return mongo_sample is not None
+
+    except Exception as e:
+        LOGGER.critical("Failed to update sample in mongo for mongo id " f"{sample_doc[FIELD_MONGODB_ID]}")
+        LOGGER.exception(e)
+
+        return False
+
+
+def check_samples_are_valid(
+    config: Config,
+    samples_collection: Collection,
+    source_plate_barcodes: List[str],
+) -> None:
+    """
+    Validate that samples do not have a sample uuid
+    """
+    samples_with_sample_uuid = list(
+        samples_collection.find(
+            {
+                FIELD_PLATE_BARCODE: {"$in": source_plate_barcodes},
+                "$and": [
+                    {FIELD_LH_SAMPLE_UUID: {"$ne": None}},
+                    {FIELD_LH_SAMPLE_UUID: {"$ne": ""}},
+                ],
+            }
+        )
+    )
+    if len(samples_with_sample_uuid) > 0:
+        raise ExceptionSampleWithSampleUUID(f"Some of the samples have a sample uuid: {samples_with_sample_uuid}")
+
+    """
+    Validate that there are matching sample rows in both mongo and mlwh for the list of barcodes supplied
+    i.e. that the sample rows are present in both databases ready to be updated
+    """
+    # select mongodb ids from mongo for barcodes list
+    query_mongo = list(
+        # fetch just the mongo ids
+        samples_collection.find({FIELD_PLATE_BARCODE: {"$in": source_plate_barcodes}}, {FIELD_MONGODB_ID: 1})
+    )
+
+    list_mongo_ids = [str(x[FIELD_MONGODB_ID]) for x in query_mongo]
+
+    if len(list_mongo_ids) == 0:
+        # There are no samples to check so this is a valid list of samples
+        return
```

Comment:
Why is a valid list of samples? It means that none of the samples for the plate barcodes provided are in mongo, but is that right? Shouldnt all the samples be always in mongo?

Reply:
Shouldnt it raise an exception instead?

Reply:
To be fair, yes we can have it raise if we prefer.  It is probably a sign that the user gave an incorrect list of barcodes to process.  There's nothing bad going to happen to the data if there are no documents to update, though.  And you'd know nothing got updated because you'd get the summary saying no documents were updated successfully and none failed.  I'll still change this to a raise, since it's more explicit about the problem.